share/completions/pkg.fish

```diff
@@ -165,3 +167,107 @@ complete -c pkg -n '__fish_pkg_is list' -xa '(pkg query "%n")'
 
 # update
 complete -c pkg -n '__fish_pkg_is add update' -s f -l force -d "Force a full download of a repository"
+
+# alias 
+function __fish_pkg_aliases
+    for alias in (pkg alias -q)
+        echo $alias | read -l name description
+
+        switch $name
+            case all-depends
+                set description 'Display all dependencies for a given package'
+                complete -c pkg -n "__fish_pkg_is $name" -xa '(pkg query "%n")'
```

Comment:
These wouldn't be called if you never try completing the subcommand before completing its argument. And if you did it multiple times you'd set up the completion multiple times, which is a bit of a waste (from a quick check it seems fish will actually run it multiple times then - we don't seem to be checking for duplicates).

I'd probably just move these out of the function with the name hardcoded - there's no real harm to setting up this completion even if the alias doesn't exist, because in that case you couldn't call it anyway.

If you want to be absolutely correct, you'd have to check `pkg alias -q` again outside of this.